src/oscar/apps/catalogue/abstract_models.py

```diff
@@ -81,6 +82,17 @@ def __str__(self):
     def has_attributes(self):
         return self.attributes.exists()
 
+    def save(self, *args, **kwargs):
+        is_product_class_has_options = self.options.count() > 0 if self.pk else False
+        super(AbstractProductClass, self).save(*args, **kwargs)
+
+        # Invalidating `has_option` field value and also value of related
+        # products of the class.
+        if is_product_class_has_options != self.has_options:
+            self.has_options = is_product_class_has_options
+            super(AbstractProductClass, self).save(*args, **kwargs)
```

Comment:
@sasha0 Do we need to call `super().save()` twice? (here and in `Product.save`)

Reply:
@craigloftus they are two separate things being dealt with. `Product` and `ProductClass` both have separate m2m relationships with `catalogue.Option`. Saving the `ProductClass` will not automatically trigger updates on associated `Product`s.